README.md
# unique_simple_bar_chart

The simple bar chart is indeed a simple design, it helps you to be able to generate a single-style simple bar chart for your design, which makes it really okay. 

You can change color or size easily anytime.


To install the package u

        unique_simple_bar_chart: ^0.0.1




To create a simple bar chart see this example with just one horizontal bar each


                  SimpleBarChart(
            
                listOfHorizontalBarData: [
                        HorizontalDetailsModel(
                        name: '1',
                        color: const Color(0xFFEB7735),
                        size: 73,
                        ),
                        HorizontalDetailsModel(
                        name: '2',
                        color: const Color(0xFFEB7735),
                        size: 92,
                        ),
                        HorizontalDetailsModel(
                        name: '3',
                        color: const Color(0xFFFBBC05),
                        size: 120,
                        ),
                         HorizontalDetailsModel(
                          name: '4',
                          color: const Color(0xFFFBBC05),
                          size: 86,
                        ),
                        HorizontalDetailsModel(
                          name: '5',
                          color: const Color(0xFFFBBC05),
                          size: 64,
                        ),
                        HorizontalDetailsModel(
                          name: '6',
                          color: const Color(0xFFFBBC05),
                          size: 155,
                        ),
                        HorizontalDetailsModel(
                          name: '7',
                          color: const Color(0xFFFBBC05),
                          size: 200,
                        ),
                        HorizontalDetailsModel(
                          name: '8',
                          color: const Color(0xFFFBBC05),
                          size: 250,
                        ),
                        HorizontalDetailsModel(
                          name: '9',
                          color: const Color(0xFFFBBC05),
                          size: 200,
                        ),
                        HorizontalDetailsModel(
                          name: '10',
                          color: const Color(0xFFFBBC05),
                          size: 160,
                        ),
                        HorizontalDetailsModel(
                          name: '11',
                          color: const Color(0xFFFBBC05),
                          size: 180,
                        ),
                        HorizontalDetailsModel(
                          name: '12',
                          color: const Color(0xFFFBBC05),
                          size: 60,
                        ),
                    ],
                verticalInterval: 100,
              ),

verticalInterval is the space between the line at the vertical size
listOfHorizontalBarData is a list of HorizontalDetailsModel, this is use to create the horizontal bars show. 

The output will be like this


<img width="440" alt="Screenshot 2023-05-20 at 20 29 55" src="https://github.com/ayanfesolution/simple_bar_chart/assets/84634788/66655266-6aac-4ed8-8a93-5ebc39611329">



To create a simple bar chart see this example with just Two horizontal bars each

        class SimpleBarChatScreen extends StatelessWidget {
          const SimpleBarChatScreen({super.key});
        
          @override
          Widget build(BuildContext context) {
            return Scaffold(
              body: Center(
                child: SimpleBarChart(
                  makeItDouble: true,
                  listOfHorizontalBarData: [
                    HorizontalDetailsModel(
                      name: 'Mon',
                      color: const Color(0xFFEB7735),
                      size: 73,
                      sizeTwo: 40,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Tues',
                      color: const Color(0xFFEB7735),
                      size: 92,
                      sizeTwo: 85,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Wed',
                      color: const Color(0xFFFBBC05),
                      size: 120,
                      sizeTwo: 100,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Thurs',
                      color: const Color(0xFFFBBC05),
                      size: 86,
                      sizeTwo: 220,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Fri',
                      color: const Color(0xFFFBBC05),
                      size: 64,
                      sizeTwo: 170,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Sat',
                      color: const Color(0xFFFBBC05),
                      size: 155,
                      sizeTwo: 120,
                      colorTwo: Colors.blue,
                    ),
                    HorizontalDetailsModel(
                      name: 'Sun',
                      color: const Color(0xFFFBBC05),
                      size: 200,
                      sizeTwo: 96,
                      colorTwo: Colors.blue,
                    ),
                  ],
                  verticalInterval: 100,
                  horizontalBarPadding: 20,
                ),
              ),
            );
          }
        }

verticalInterval is the space between the line at the vertical size
listOfHorizontalBarData is a list of HorizontalDetailsModel, this is used to create the horizontal bars shown. 
horizontalBarPadding is to adjust the spacing between one bar to another, in case u experience overflow, the default size is 22
makeItDouble is set to true if the bars are to be two, its default value is false       



<img width="279" alt="Screenshot 2023-08-01 at 12 24 33" src="https://github.com/ayanfesolution/simple_bar_chart/assets/84634788/d7fd96c5-8a47-47e3-8d96-0878c171e8f1">





CHANGELOG.md
## 0.0.1

* Design a simple style simple bar change with minimum required field.

## 0.0.2

* Added example folder and also fix bugs


## 0.0.4

* Overflow bug fixed
* Make adjustment for two horizontal bars
* Add more customization options to the widget like color, size etc..



lib/horizontal_bar.dart
import 'package:flutter/material.dart';

class HorizontalBar extends StatelessWidget {
  const HorizontalBar({
    super.key,
    required this.verticalBarTextStyleColor,
    required this.name,
    required this.barColor,
    this.barColorTwo,
    required this.barSize,
    required this.barInterver,
    this.isItDouble = false,
    required this.barSizeTwo,
    required this.horizontalBarPadding,
  });

  final Color verticalBarTextStyleColor;
  final String name;
  final Color barColor;
  final Color? barColorTwo;
  final double barSize, barInterver, horizontalBarPadding;
  final bool isItDouble;
  final double barSizeTwo;

  @override
  Widget build(BuildContext context) {
    return isItDouble
        ? Padding(
            padding: EdgeInsets.symmetric(
                horizontal: (horizontalBarPadding - 9), vertical: 0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                Row(
                  crossAxisAlignment: CrossAxisAlignment.end,
                  children: [
                    Container(
                      height: (52.5) * (barSize / barInterver),
                      width: 9,
                      color: barColor,
                    ),
                    Container(
                      height: (52.5) * (barSizeTwo / barInterver),
                      width: 9,
                      color: barColorTwo,
                    ),
                  ],
                ),
                SizedBox(
                  height: 25,
                  child: Padding(
                    padding: const EdgeInsets.only(top: 8),
                    child: Text(
                      name,
                      style: TextStyle(
                        color: verticalBarTextStyleColor,
                        fontSize: 10,
                        fontWeight: FontWeight.w500,
                      ),
                    ),
                  ),
                ),
              ],
            ),
          )
        : Padding(
            padding: EdgeInsets.symmetric(horizontal: horizontalBarPadding),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                Container(
                  height: (52.5) * (barSize / barInterver),
                  width: 9,
                  color: barColor,
                ),
                SizedBox(
                  height: 25,
                  child: Padding(
                    padding: const EdgeInsets.only(top: 8),
                    child: Text(
                      name,
                      style: TextStyle(
                        color: verticalBarTextStyleColor,
                        fontSize: 10,
                        fontWeight: FontWeight.w500,
                      ),
                    ),
                  ),
                ),
              ],
            ),
          );
  }
}


lib/horizontal_line.dart
import 'package:flutter/material.dart';

class SimpleBarChartHorizotalLine extends StatelessWidget {
  const SimpleBarChartHorizotalLine({
    super.key,
    required this.verticalBarTextStyleColor,
    required this.horizontalLineColor,
    required this.lineNumber,
  });

  final Color verticalBarTextStyleColor;
  final Color horizontalLineColor;
  final double lineNumber;

  @override
  Widget build(BuildContext context) {
    return Row(
      crossAxisAlignment: CrossAxisAlignment.center,
      children: [
        Text(
          '$lineNumber',
          style: TextStyle(
            fontSize: 10,
            fontWeight: FontWeight.w500,
            color: verticalBarTextStyleColor,
          ),
        ),
        Expanded(
          child: Container(
            height: 1,
            width: double.maxFinite,
            color: horizontalLineColor,
          ),
        )
      ],
    );
  }
}


lib/simple_bar_chart.dart
library unique_simple_bar_chart;

import 'package:flutter/material.dart';
import 'data_models.dart';
import 'horizontal_bar.dart';
import 'horizontal_line.dart';

class SimpleBarChart extends StatefulWidget {
  const SimpleBarChart({
    super.key,
    this.fullBarChartHeight = 216,
    this.verticalBarTextStyleColor = const Color(0xFF979797),
    this.horizontalLineColor = const Color(0xFFC8C8C8),
    required this.verticalInterval,
    required this.listOfHorizontalBarData,
    this.makeItDouble = false,
    this.horizontalBarPadding = 22,
  });
  final double fullBarChartHeight;
  final Color verticalBarTextStyleColor;
  final Color horizontalLineColor;
  final bool makeItDouble;

  final double verticalInterval, horizontalBarPadding;

  final List<HorizontalDetailsModel> listOfHorizontalBarData;

  @override
  State<SimpleBarChart> createState() => _SimpleBarChartState();
}

class _SimpleBarChartState extends State<SimpleBarChart> {
  @override
  Widget build(BuildContext context) {
    return SizedBox(
      height: widget.fullBarChartHeight,
      width: double.maxFinite,
      child: Stack(
        children: [
          Column(
            mainAxisAlignment: MainAxisAlignment.end,
            children: [
              for (var i = 0; i < 4; i++)
                Padding(
                  padding: const EdgeInsets.symmetric(vertical: 20),
                  child: SimpleBarChartHorizotalLine(
                    verticalBarTextStyleColor: widget.verticalBarTextStyleColor,
                    horizontalLineColor: widget.horizontalLineColor,
                    lineNumber: ((widget.verticalInterval * 3) -
                        (widget.verticalInterval * i)),
                  ),
                ),
            ],
          ),
          Row(
            children: [
              const SizedBox(
                width: 30,
              ),
              for (var i = 0; i < widget.listOfHorizontalBarData.length; i++)
                HorizontalBar(
                  verticalBarTextStyleColor: widget.verticalBarTextStyleColor,
                  name: widget.listOfHorizontalBarData[i].name,
                  barColor: widget.listOfHorizontalBarData[i].color,
                  barSize: widget.listOfHorizontalBarData[i].size,
                  barInterver: widget.verticalInterval,
                  isItDouble: widget.makeItDouble,
                  barColorTwo: widget.listOfHorizontalBarData[i].colorTwo,
                  barSizeTwo: widget.listOfHorizontalBarData[i].sizeTwo ?? 0,
                  horizontalBarPadding: widget.horizontalBarPadding,
                ),
            ],
          ),
        ],
      ),
    );
  }
}


example/example.dart
import 'package:flutter/material.dart';
import 'package:unique_simple_bar_chart/data_models.dart';
import 'package:unique_simple_bar_chart/simple_bar_chart.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: SimpleBarChatScreen(),
    );
  }
}

class SimpleBarChatScreen extends StatelessWidget {
  const SimpleBarChatScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: SimpleBarChart(
          listOfHorizontalBarData: [
            HorizontalDetailsModel(
              name: '1',
              color: const Color(0xFFEB7735),
              size: 73,
            ),
            HorizontalDetailsModel(
              name: '2',
              color: const Color(0xFFEB7735),
              size: 92,
            ),
            HorizontalDetailsModel(
              name: '3',
              color: const Color(0xFFFBBC05),
              size: 120,
            ),
            HorizontalDetailsModel(
              name: '4',
              color: const Color(0xFFFBBC05),
              size: 86,
            ),
            HorizontalDetailsModel(
              name: '5',
              color: const Color(0xFFFBBC05),
              size: 64,
            ),
            HorizontalDetailsModel(
              name: '6',
              color: const Color(0xFFFBBC05),
              size: 155,
            ),
            HorizontalDetailsModel(
              name: '7',
              color: const Color(0xFFFBBC05),
              size: 200,
            ),
            HorizontalDetailsModel(
              name: '8',
              color: const Color(0xFFFBBC05),
              size: 250,
            ),
            HorizontalDetailsModel(
              name: '9',
              color: const Color(0xFFFBBC05),
              size: 200,
            ),
            HorizontalDetailsModel(
              name: '10',
              color: const Color(0xFFFBBC05),
              size: 160,
            ),
            HorizontalDetailsModel(
              name: '11',
              color: const Color(0xFFFBBC05),
              size: 180,
            ),
            HorizontalDetailsModel(
              name: '12',
              color: const Color(0xFFFBBC05),
              size: 60,
            ),
          ],
          verticalInterval: 100,
        ),
      ),
    );
  }
}

class SimpleBarChatScreenTwo extends StatelessWidget {
  const SimpleBarChatScreenTwo({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: SimpleBarChart(
          makeItDouble: true,
          listOfHorizontalBarData: [
            HorizontalDetailsModel(
              name: 'Mon',
              color: const Color(0xFFEB7735),
              size: 73,
              sizeTwo: 40,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Tues',
              color: const Color(0xFFEB7735),
              size: 92,
              sizeTwo: 85,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Wed',
              color: const Color(0xFFFBBC05),
              size: 120,
              sizeTwo: 100,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Thurs',
              color: const Color(0xFFFBBC05),
              size: 86,
              sizeTwo: 220,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Fri',
              color: const Color(0xFFFBBC05),
              size: 64,
              sizeTwo: 170,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Sat',
              color: const Color(0xFFFBBC05),
              size: 155,
              sizeTwo: 120,
              colorTwo: Colors.blue,
            ),
            HorizontalDetailsModel(
              name: 'Sun',
              color: const Color(0xFFFBBC05),
              size: 200,
              sizeTwo: 96,
              colorTwo: Colors.blue,
            ),
          ],
          verticalInterval: 100,
          horizontalBarPadding: 20,
        ),
      ),
    );
  }
}
